src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/services/RadicleApplicationService.java

```diff
@@ -39,42 +35,38 @@ public RadicleApplicationService(RadicleSettingsHandler radicleSettingsHandler)
         this.settingsHandler = radicleSettingsHandler;
     }
 
-    public void setRadConfigPaths(String gitStoragePath, String keysStoragePath) {
-        if (radStoragePath == null) {
-            radStoragePath = new HashMap<>();
-        }
-        radStoragePath.put("gitStoragePath", gitStoragePath);
-        radStoragePath.put("keysStoragePath", keysStoragePath);
+    public ProcessOutput homePath() {
+        return executeCommand(".", List.of("path"), null);
     }
 
-    public RadConfig getRadStoragePath() {
-        if (radStoragePath == null) {
-            return null;
-        }
-        return new RadConfig(radStoragePath.get("gitStoragePath"), radStoragePath.get("keysStoragePath"));
+    public ProcessOutput radPath() {
+        return executeCommand("which", ".", List.of("rad"), null);
     }
 
-    public ProcessOutput getRadPath() {
-        return executeCommand("which", ".", List.of("rad"), null, false);
-    }
-
-    public ProcessOutput self(boolean activeProfile) {
-        if (activeProfile) {
-            return executeCommand(".", List.of("self", "--profile"), null);
-        } else {
-            return executeCommand(".", List.of("self"), null);
-        }
+    public ProcessOutput self(String radHome, String radPath) {
+        final var settings = settingsHandler.loadSettings();
+        final var radExePath = Strings.isNullOrEmpty(radPath) ? settings.getPath() : radPath;
+        final var radHomePath = Strings.isNullOrEmpty(radHome) ? settings.getRadHome() : radHome;
+        return executeCommand(radExePath, radHomePath, ".", List.of("self"), null, "");
     }
 
     public ProcessOutput clone(String urn, String directory) {
         return executeCommand(directory, List.of("clone", urn, "--no-confirm"), null);
     }
 
+    public boolean isIdentityUnlocked(String key) {
+        var output = executeCommand("ssh-add", "", ".", List.of("-l"), null, "");
+        if (!RadAction.isSuccess(output) || Strings.isNullOrEmpty(key)) {
```

Comment:
if the key is empty, then it's better to return false early and skip executing the command